### PR TITLE
Create item search

### PR DESCRIPTION
**Pull request by Arturnix**
Added search bar with logic to search items.

### CODE REVIEW

File: src/main/java/pl/arturzgodka/apihandlers/BaseUrlParts.java [hunk 1]
```diff
@@ -10,5 +10,5 @@ public class BaseUrlParts {
     public static final String BASE_SKILL_API = "/skill/";
     public static final String BASE_ITEM_API = BASE_BLIZZARD_URL + "data/item/";
     public static final String BASE_DATA_HERO_API = BASE_BLIZZARD_URL + "data/hero/";
-    public static final String BASE_LOCALE_AND_TOKEN = "/?locale=pl_PL&access_token=";
+    public static final String BASE_LOCALE_AND_TOKEN = "/?locale=en_US&access_token=";
```
[the remark this answers is not in the document]
Comment (Arturnix) [reply src/main/java/pl/arturzgodka/apihandlers/BaseUrlParts.java:13]: Changed for en_EU locale. Done

File: src/main/java/pl/arturzgodka/controllers/ItemsController.java [hunk 2]
```diff
@@ -26,6 +28,29 @@ public String getItemTypes(Model model) {
         return "itemsTypes";
     }
 
+    @GetMapping("/singleItem")
+    public String searchItemForm(Model model, @RequestParam String itemSearchName) {
+
+        //TODO ujednolicic mapowanie obiektow i view dla nich czyli niech bedzie zwracana jedna templatka i dac pola tylko opcjonalnie w zaleznosci jakiego typu obiekt zostal zwrocony.
```
[the remark this answers is not in the document]
Comment (Arturnix) [reply src/main/java/pl/arturzgodka/controllers/ItemsController.java:34]: TODO done

File: src/main/java/pl/arturzgodka/controllers/ItemsNamesRepository.java [hunk 3]
```diff
@@ -0,0 +1,22 @@
+package pl.arturzgodka.controllers;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class ItemsNamesRepository {
+
+    private final static Map<String, String> itemsNamesAndSlugs = new HashMap<String, String>() {{
```
[the remark this answers is not in the document]
Comment (Arturnix) [reply src/main/java/pl/arturzgodka/controllers/ItemsNamesRepository.java:8]: For present architecture of code, this class is not useful - delete. Done

File: src/main/resources/templates/items.html [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+    <link
+            rel="stylesheet"
+            href="https://maxcdn.bootstrapcdn.com/bootstrap/4.0.0/css/bootstrap.min.css"
+            integrity="sha384-Gn5384xqQ1aoWXA+058RXPxPg6fy4IWvTNh0E263XmFcJlSAwiGgFAW/dAiS6JXm"
+            crossorigin="anonymous"
+    />
+    <link rel="stylesheet" href="/css/styles.css"/>
+
+    <!-- Font Awesome -->
+    <script
```
[the remark this answers is not in the document]
Comment (Arturnix) [reply src/main/resources/templates/items.html:17]: This font is not used in view - delete. Done

File: src/main/java/pl/arturzgodka/controllers/ItemClassesAndNamesLists.java [hunk 5]
```diff
@@ -140,6 +141,24 @@ public static List<List<String>> getArmorSelectedItemFullLists(String itemType,
         return itemsTypesMap.get(itemType).get(selectedItem);
     }
 
+    public static List<List<String>> selectedItemNameToApi(String searchItemName) {
+
+        List<String> itemsTypesMapKeys = itemsTypesMap.keySet().stream().toList();
+        List<List<String>> itemsMatch = new ArrayList<>();
+
+        for(String mainKey : itemsTypesMapKeys) {
```
[the remark this answers is not in the document]
Comment (Arturnix) [reply src/main/java/pl/arturzgodka/controllers/ItemClassesAndNamesLists.java:149]: Code refactored. Done

**Review by blazejprzyluski — APPROVED** [review]

**Pull request merged**